tests/migration/postcopy.go

```diff
@@ -168,39 +168,32 @@ var _ = SIGMigrationDescribe("VM Post Copy Live Migration", func() {
 		)
 
 		Context("and fail", Serial, func() {
-			var createdPods []string
-			BeforeEach(func() {
-				createdPods = []string{}
-			})
+			var killerPod string
 
-			runMigrationKillerPod := func(nodeName string) {
-				podName := fmt.Sprintf("migration-killer-pod-%s", rand.String(5))
+			runVirtHandlerKillerPod := func(nodeName string) {
+				podName := "migration-killer-pod-"
 
 				// kill the handler
 				pod := libpod.RenderPrivilegedPod(podName, []string{"/bin/bash", "-c"}, []string{fmt.Sprintf("while true; do pkill -9 virt-handler && sleep 5; done")})
 
 				pod.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeName}
 				createdPod, err := virtClient.CoreV1().Pods(pod.Namespace).Create(context.Background(), pod, metav1.CreateOptions{})
-				Expect(err).ToNot(HaveOccurred(), "Should create helper pod")
-				createdPods = append(createdPods, createdPod.Name)
-				Expect(createdPods).ToNot(BeEmpty(), "There is no node for migration")
+				Expect(err).ToNot(HaveOccurred(), "Should create killer pod")
+				killerPod = createdPod.Name
 			}
 
-			removeMigrationKillerPod := func() {
-				for _, podName := range createdPods {
-					Eventually(func() error {
-						err := virtClient.CoreV1().Pods(testsuite.NamespacePrivileged).Delete(context.Background(), podName, metav1.DeleteOptions{})
-						return err
-					}, 10*time.Second, 1*time.Second).Should(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"), "Should delete helper pod")
+			removeVirtHandlerKillerPod := func() {
+				Expect(killerPod).NotTo(BeEmpty())
+				Eventually(func() error {
+					err := virtClient.CoreV1().Pods(testsuite.NamespacePrivileged).Delete(context.Background(), killerPod, metav1.DeleteOptions{})
+					return err
+				}, 10*time.Second, 1*time.Second).Should(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"), "Should delete helper pod")
 
-					Eventually(func() error {
-						_, err := virtClient.CoreV1().Pods(testsuite.NamespacePrivileged).Get(context.Background(), podName, metav1.GetOptions{})
-						return err
-					}, 5*time.Minute, 1*time.Second).Should(
-						SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())),
-						"The killer pod should be gone within the given timeout",
-					)
-				}
+				Eventually(func() error {
+					_, err := virtClient.CoreV1().Pods(testsuite.NamespacePrivileged).Get(context.Background(), killerPod, metav1.GetOptions{})
+					return err
+				}, 5*time.Minute, 1*time.Second).Should(
+					MatchError(errors.IsNotFound, "k8serrors.IsNotFound"), "The killer pod should be gone within the given timeout")
```

Comment:
I am not sure this makes sense. If the Eventually block above succeded, it means that the killerPod is already gone

Reply:
I agree! This is untouched from the original version, I thought maybe there was some weird reason for it, but you're right I'll just nuke it.